### ISSUE TITLE
Audio track enhancement

### ISSUE THREAD

**Issue by cmanucy** [smeighan/xLights#2715]
An idea regarding timing tracks vs. on-the-fly audio analysis:  there are some tools (and open source projects like deezer/spleeter) that can analyze audio tracks and segregate the drums, bass, vocals from the rest of the track.  The concept is to use one of these kinds of plugins so you could reference that as a separate track in xLights - similar to a timing track, but this would not be editable.

Once analyzed, you could point an effect like VU Meter to use that sub-track for its on-beat hits or whatever.

Currently in order to accomplish this same goal, one would have to take the audio, split it, then analyze it separately and bring it back in as a timing track.  With something integrated, all of that work would be consolidated - however I see it could create a lot more effort on trying to display it visually.

**Comment by deplanche** [comment]
If you right click on the waveform in xlights, you can already do several of these options, to make different timing tracks.

**Comment by cmanucy** [comment]
Thank you for reminding me - I had forgotten about that.  But after refreshing my memory on it and playing with it, there are a couple of items that stand out as enhancements in my eyes.  One, you can't reference those modifications.  So one would have to create a manual timing track off of the modified waveform, which isn't what I'm after.  I was hoping to be able to use something similar directly (or 'immediately') on an effect.

Two, the version built into xLights is a bit terrible compared to some of the other applications I've used (including OSS ones).  

That's where I'm seeing a delta between what's available and the enhancement request.

**Comment by keithsw1111** [comment]
If you want this you are going to need to do some of the leg work ... what exactly are you looking for and where in those large code bases are the relevant components.

**Comment by sslupsky** [comment]
This is a very good idea.  My suggestion is the spleeter tracks could replace the tracks that are displayed using the right click on the waveform in xLights.  As @cmanucy suggested it would also be useful to have the ability to have multiple audio tracks and allow the effects to choose a track.

The audio separation is quite amazing.

**Comment by fone626** [comment]
This ticket made me look into spleeter as I had never heard of it and it immediately became part of my work flow in sequencing songs.  It would be amazing to have it included directly into xLights.
If xLights could possibly get the option to quickly switch between any audio tracks inside a directory with the same name (minus the extension) as the song itself.  That would allow us to use spleeter, or any other pre-processor and use it inside xlights without having to immediately add support directly into xLights itself.
Right now I have to go into the sequence setting and constantly switch between the spleeter output files.

**Comment by sslupsky** [comment]
Yeah, it's an incredibly useful tool.  That's a good suggestion on being able to cycle through the tracks.  I starting working with the stems in Audacity and working with labeling.  Using the Label Sounds tool (Analyze menu) you can automate the generation of beats for the drum stem.  But I am finding it difficult to be productive with Audacity labeling because I find Audacity's UI is very weak with tweaking the labels.

**Comment by derwin12** [comment]
This will be bundled into another ticket..
